Quadriga/src/main/java/edu/asu/spring/quadriga/rest/SearchNetworkRestController.java

```diff
@@ -0,0 +1,128 @@
+    
+    /**
+     * Rest interface to search concept in list of projects
+     * http://<<URL>:<PORT>>/quadriga/rest/networks/search/{conceptId}/{projectIds}
+     * http://localhost:8080/quadriga/rest/networks/search?conceptId=http://www.digitalhps.org/concepts/CON516ec8c2-20bc-4f35-ae1e-ad908db9d662&projectIds=PROJfFnfnn,PROJFgGPpk
+     * 
+     * @param conceptId: ConceptId for searching node
+     * @param projectIds : List of comma separated projectId
+     * @param response
+     * @param accept
+     * @param principal
+     * @return status
+     * @throws RestException
+     */
+    
+    @RequestMapping(value = "rest/network/search", method = RequestMethod.GET, produces = "application/xml")
+    public ResponseEntity<String> getSearchTransformedNetwork(@RequestParam("conceptId") String conceptId, @RequestParam("projectIds") List<String> projectIds,
+            HttpServletResponse response, String accept, Principal principal, HttpServletRequest req) throws RestException {
+    	
+    	List<IProject> projectList = new ArrayList<>();
+    	
+    	try {    		
+    	    for(String projectId : projectIds){
+    	    	IProject project = null;
+    	    	try{
+                    project = projectManager.getProjectDetails(projectId);
+    	    	}catch(QuadrigaStorageException e){
+    	    		logger.error("QuadrigaStorageException:", e);
+    	    	}
+                if(project!=null){
+                    projectList.add(project);
+                }
+            }
+    	    
+    		
+            if(projectList.size()==0){
+                String errorMsg = errorMessageRest.getErrorMsg("Projects don't exist.");
+                return new ResponseEntity<String>(errorMsg, HttpStatus.NOT_FOUND);
+            }
+
+            ITransformedNetwork transformedNetwork;
+            try{
+                transformedNetwork = transformationManager.getSearchTransformedNetworkMultipleProjects(
+        			projectIds, conceptId);	
+            }catch(QuadrigaStorageException e){
+                throw new RestException(403, e);
+	    	}
+            
+            
+    	    VelocityEngine engine = restVelocityFactory.getVelocityEngine(req);
+            engine.init();
+            Template template = engine.getTemplate("velocitytemplates/transformationdetails.vm");
+            VelocityContext context = new VelocityContext(restVelocityFactory.getVelocityContext());
+            context.put("nodeList", new ArrayList<Node>(transformedNetwork.getNodes().values()));
+            context.put("linkList", transformedNetwork.getLinks());
+            context.put("graphId", conceptId);
+            
+            StringWriter writer = new StringWriter();
+            template.merge(context, writer);
+            HttpHeaders httpHeaders = new HttpHeaders();
+            httpHeaders.setContentType(MediaType.APPLICATION_XML);
+            return new ResponseEntity<String>(writer.toString(), httpHeaders, HttpStatus.OK);
+        }catch (Exception e) {

```

Comment:
Never have a catch all block like this. Catch the exceptions exactly where they are thrown. What are you catching here? Put it only around the code that throws it.